### PR TITLE
implement fake guest thread in client/server live migration sample

### PR DESCRIPTION
**Pull request by tmakatos**

### CODE REVIEW

File: samples/client.c [hunk 1]
```diff
@@ -806,22 +807,69 @@ do_migrate(int sock, int migr_reg_index, size_t nr_iters,
...
 }
 
+struct fake_guest_data {
+    int sock;
+    size_t bar1_size;
+    bool done;
+    unsigned char *md5sum;
+};
+
+static void*
```
Comment (jlevon) [inline samples/client.c:817]: nit, "void *"

File: samples/client.c [hunk 2]
```diff
@@ -806,22 +807,69 @@ do_migrate(int sock, int migr_reg_index, size_t nr_iters,
...
+fake_guest(void *arg)
+{
+    struct fake_guest_data *fake_guest_data = arg;
+    int ret;
+    char buf[fake_guest_data->bar1_size];
+	MD5_CTX md5_ctx;
+    FILE *fp = fopen("/dev/urandom", "r");
+
+    if (fp == NULL) {
+        err(EXIT_FAILURE, "failed to open /dev/unrandom");
```
Comment (jlevon) [inline samples/client.c:827]: nit, "urandom"

**Review by jlevon — CHANGES_REQUESTED** [review]
just a couple of nits

File: samples/CMakeLists.txt [hunk 3]
```diff
@@ -30,7 +30,7 @@
 
 add_executable(client client.c
                ../lib/tran_sock.c ../lib/migration.c)
-target_link_libraries(client json-c)
+target_link_libraries(client json-c pthread ssl crypto)
```
[the remark this answers is not in the document]
Comment (tmakatos) [reply samples/CMakeLists.txt:33]: ok

[the remark this answers is not in the document]
Comment (tmakatos) [reply samples/CMakeLists.txt:33]: Actually they're provided by `libssl-dev`, which we already require.

File: samples/client.c (hunk 1 above)
Comment (tmakatos) [reply samples/client.c:817]: ok

File: samples/client.c (hunk 2 above)
Comment (tmakatos) [reply samples/client.c:827]: ok

**Review by jlevon — APPROVED** [review]

**Pull request merged**